aacparser.go

```diff
@@ -2128,25 +2147,37 @@ func (adts *ADTS) sbr_channel_pair_element(ext_data *sbr_extension_data, bs_amp_
 
 		// TODO: could we be a bit more graceful about this block?
 		num_bits_left := cnt * 8
-		e.Bs_extension_id = make([]uint8, 0)
-		e.Sbr_extension = make([]*sbr_extension, 0)
-		for i := 0; num_bits_left > 7; i++ {
-			ext_id, _ := adts.reader.ReadBitsAsUInt8(2)
-			e.Bs_extension_id = append(e.Bs_extension_id, ext_id)
-			if e.Bs_extension_id[i] == EXTENSION_ID_PS {
-				num_bits_left -= 2
-
-				bits_read, ext := adts.sbr_extension(e.Bs_extension_id[i], num_bits_left)
-				e.Sbr_extension = append(e.Sbr_extension, ext)
-
-				num_bits_left -= bits_read
-				if num_bits_left < 0 {
-					return e, fmt.Errorf("Error: SBR parsing overran available bits")
-				}
-			}
-		}
 
-		e.Bs_fill_bits, _ = adts.reader.ReadBitsToByteArray(num_bits_left)
+		if num_bits_left > MaxBitsLeft {
+			return e, fmt.Errorf("Too many bits left, check bitstream continuity")
+		}
+
+		// TODO: Uncomment this when extensions are supported.
```

Comment:
egh no, just remove the non-functional code rather than comment it out. Commented out code is more of a liability than a benefit to future users/readers/implementers.